cpp/include/cudf/datetime.hpp

```diff
@@ -38,6 +38,23 @@ namespace datetime {
  * @file
  */
 
+/**
+ * @brief Types of datetime components that may be extracted.
+ */
+enum class datetime_component : uint8_t {
+  INVALID = 0,
```

Comment:
How is INVALID used? Now that this is becoming a public enum, it needs higher scrutiny.

Reply:
Let’s remove it. I think that it is unnecessary. We do not have this kind of pattern in most of our enums.